tests/e2e/helpers.go

```diff
@@ -2082,6 +2093,50 @@ func waitForPod(podCallback wait.ConditionFunc) error {
 	return wait.PollImmediate(retryInterval, timeout, podCallback)
 }
 
+// check if pod name has priority class set to the given value.
+func checkPodPriorityClass(t *testing.T, c kubernetes.Interface, podName, namespace, priorityClass string) wait.ConditionFunc {
+	return func() (bool, error) {
+		pod, err := c.CoreV1().Pods(namespace).Get(goctx.TODO(), podName, metav1.GetOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return false, err
+		}
+
+		if apierrors.IsNotFound(err) {
+			E2ELogf(t, "Pod %s not found yet", podName)
+			return false, nil
+		}
+
+		if pod.Spec.PriorityClassName != priorityClass {
+			E2ELogf(t, "pod %s has priority class %s, expected %s", podName, pod.Spec.PriorityClassName, priorityClass)
+			return true, nil
+		}
+
+		return true, nil
+	}
+}
+
+// check if pod has empty priority class
+func checkPodEmptyPriorityClass(t *testing.T, c kubernetes.Interface, podName, namespace string) wait.ConditionFunc {
```

Comment:
Couldn't we collapse these two assertions into a single function?

Do we need a separate function for checking empty pod priority classes if we have the ability to pass in "" as the priorityClass?

```go
// make sure the priority class matches one of the defaults
expectedPriorityClass := "system-cluster-critical"
waitForPod(checkPodPriorityClass(t, f.KubeClient, pod.Name, namespace, expectedPriorityClass))
// make sure the priority class is empty
expectedPriorityClass := ""
waitForPod(checkPodPriorityClass(t, f.KubeClient, pod.Name, namespace, expectedPriorityClass))
```

Might save us from having a separate, but nearly identical method.